src/tardev-snapshotter/src/snapshotter.rs

```diff
@@ -856,40 +856,64 @@ impl TarDevSnapshotter {
                     "missing target layer digest label",
                 ));
             };
+
+            // Determine image layer media type
             let Some(media_type) = labels.get(TARGET_LAYER_MEDIA_TYPE_LABEL) else {
                 error!("missing target layer media type label");
                 return Err(Status::invalid_argument(
                     "missing target layer media type label",
                 ));
             };
-
             trace!("layer digest {} media_type: {:?}", digest_str, media_type);
 
-            // TODO use the media_type:
-            // gzipped: "application/vnd.docker.image.rootfs.diff.tar.gzip", "application/vnd.oci.image.layer.v1.tar+gzip";
-            // uncompressed: "application/vnd.oci.image.layer.v1.tar", "application/vnd.docker.image.rootfs.diff.tar"
+            let layer_type = match media_type.as_str() {
+                "application/vnd.docker.image.rootfs.diff.tar.gzip" 
+                | "application/vnd.oci.image.layer.v1.tar+gzip" => "tar.gz",
+                "application/vnd.oci.image.layer.v1.tar" 
+                | "application/vnd.docker.image.rootfs.diff.tar" => "tar",
+                _ => {
+                    error!("unsupported media type: {}", media_type);
+                    return Err(Status::invalid_argument(format!(
+                        "unsupported media type: {}",
+                        media_type
+                    )));
+                }
+            };
 
+            // Fetch the layer image
             let name = dir.path().join(name_to_hash(&key));
```

Comment:
Can we add the extension right away and not have to rename later?

Reply:
Good catch. There is no need to rename later. Simplified this code section